core/video_options.hpp

```diff
@@ -144,6 +144,13 @@ struct VideoOptions : public Options
 		if ((split || segment) && output.find('%') == std::string::npos)
 			LOG_ERROR("WARNING: expected % directive in output filename");
 
+		// From https://en.wikipedia.org/wiki/Advanced_Video_Coding#Levels
+		if (width * height * framerate.value_or(DEFAULT_FRAMERATE) / (16 * 16) > 245760)
```

Comment:
Rounding means that this is wrong.

1920x1080 is 120x68 = 8160 per frame (1080 not being a multiple of 16).
Using a bare 1920x1080 gives me a max frame rate of 30.34fps, when the actual limit is 30.11fps.

Reply:
And only if level is not explicitly set by the user?

Reply:
I made this explicit as if the user sets 4.1 with a rate that overflows the limit, it won't work. That brings us back to square one.

Reply:
Seeing as level 4.2 is the max that we can realistically expect to achieve in real time, this seems reasonable.

It does mean the app will be a little noisy if --level 4.2 has been explicitly specified on the command line though.

Reply:
> Rounding means that this is wrong.
> 
> 1920x1080 is 120x68 = 8160 per frame (1080 not being a multiple of 16). Using a bare 1920x1080 gives me a max frame rate of 30.34fps, when the actual limit is 30.11fps.

Given that framerate is a flot, rounding should be handled correctly right? Unless I'm missing something obvious...

Reply:
> Seeing as level 4.2 is the max that we can realistically expect to achieve in real time, this seems reasonable.
> 
> It does mean the app will be a little noisy if --level 4.2 has been explicitly specified on the command line though.

I wonder if we should test the upper limit and bomb out as well?

Reply:
> Given that framerate is a float, rounding should be handled correctly right? Unless I'm missing something obvious..

It's a macroblock limit, not pixel limit. A macroblock is 16x16 pixels and you can't encode fractional MBs.

Both width and height need to be rounded up to a multiple of 16 before multiplication.